### DISCUSSION TITLE
Add more fields to the recipe actions

### DISCUSSION THREAD

**Discussion by felixschndr** [mealie-recipes/mealie#3610] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

The ability to `link` and `post` recipes to certain weburls was added in https://github.com/mealie-recipes/mealie/pull/3448. As discussed [here](https://github.com/mealie-recipes/mealie/pull/3448#issuecomment-2115936511) this is the feature request to add two more fields to the `valid merge fields`

### Please provide a concise description of the feature that would resolve your issue.

Please add the fields `baseQuantity` and `requestedQuantity` to the `valid merge fields`

### Please consider and list out some caveats or tradeoffs made in your design decision

None

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [X] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by felixschndr** [comment]
@michael-genson Would you be able to implement this as a followup to [your comment in #3448](https://github.com/mealie-recipes/mealie/pull/3448#issuecomment-2115939812)? That would be great, thank you :)

**Reply by michael-genson** [reply]
Yeah I plan to tackle this at some point. Unfortunately it's not as trivial as exposing that as a merge field since our "yield" value is unstructured

**Reply by felixschndr** [reply]
Hey @michael-genson, any news on this? (Background https://github.com/mealie-recipes/mealie/pull/4489 and https://github.com/felixschndr/mealie-bring-api/issues/15)

**Reply by michael-genson** [reply]
Now that https://github.com/mealie-recipes/mealie/pull/4489 is merged this should be rather easy. I should hopefully be able to it soon! Thanks for the reminder

**Comment by michael-genson** [comment]
FYI: https://github.com/mealie-recipes/mealie/pull/4952

**Reply by felixschndr** [reply]
Thank you very much. I didn't understand the difference between `servings` and `yields` yet. Can you point me to somewhere where this is explained please?

**Reply by michael-genson** [reply]
Servings are for how many people a recipe serves, whereas yield is for how much of something is made. Most recipes only use one or the other, but a good example would be something like meatballs: this recipe yields 24 meatballs, which serves 8 people.

**Reply by michael-genson** [reply]
For the purpose of Bring, you probably want servings

**Reply by felixschndr** [reply]
Thanks for the explanation, that helps a lot!

**Reply by felixschndr** [reply]
@michael-genson 
I just installed Mealie `v2.6`. Thank you for the new release!

----

However, I am having a bit of a difficult time to integrate the new fields into my integration since it relies on the `post` action instead of the `link` action.
1. I was looking into using the `link` action instead. However I do not understand how this works. The browser sends a GET request to the specified target. In this URL placeholders can be added (for example the URL of the recipe) (e.g. `https://api.getbring.com/rest/bringrecipes/deeplink?url=${url}&source=web
`). How does bring then access the data of the recipe?
  a. Assuming the mealie instance is publicly available, how is bring able to access the recipe as it is not logged in?
  b. What does bring to with the data it receives as if it were logged in it would receive html code, right? Do they have an internal parser that works with the HTML?
2. It looks like relying on the `link` action does not work for my integration as I'd like to work with the `JSON` of the recipe. Would it be possible to add the newly created fields as placeholders to the `post` action as well? That'd be great!
  If I see this correctly, this would only require to pass in the url parameter [here](https://github.com/mealie-recipes/mealie/blob/mealie-next/frontend/composables/use-group-recipe-actions.ts#L72) and use it [here](https://github.com/mealie-recipes/mealie/blob/mealie-next/frontend/lib/api/user/group-recipe-actions.ts#L17), right?

Thank you very much in advance.

**Reply by michael-genson** [reply]
> How does bring then access the data of the recipe?

Bring provides that URL to import recipes via a bookmark. They basically scrape the webpage (similar to how Mealie does it). Mealie exposes the recipe data in `schema.org` format, so it's scrape-able.

> Would it be possible to add the newly created fields as placeholders to the post action as well?

`POST` actions include the entire recipe JSON in the body, so you can use any recipe fields you want.

**Reply by felixschndr** [reply]
>> Would it be possible to add the newly created fields as placeholders to the post action as well?

> `POST` actions include the entire recipe JSON in the body, so you can use any recipe fields you want.

However, this only includes the recipe JSON and not the amount of servings/yields and the quantity of the `food`s is not scaled according to the chosen amount of servings/yields. This it would be necessary to add the placeholders to the `post` action as well

**Reply by felixschndr** [reply]
@michael-genson any thoughts on this? 
(sorry for the ping)

**Reply by felixschndr** [reply]
I took a closer look at the source code. It seems like my original idea on how to fix this (2. part of https://github.com/mealie-recipes/mealie/discussions/3610#discussioncomment-12094439) was completely wrong.

The client (browser) sends a get a post request to the server (triggered [here](https://github.com/mealie-recipes/mealie/blob/mealie-next/frontend/composables/use-group-recipe-actions.ts#L72) and sent [here](https://github.com/mealie-recipes/mealie/blob/mealie-next/frontend/lib/api/user/group-recipe-actions.ts#L17)).
The server (backend) gets the request and handles is [here](https://github.com/mealie-recipes/mealie/blob/mealie-next/mealie/routes/households/controller_group_recipe_actions.py#L71). It dumps the the recipe to json and sends the request [here](https://github.com/mealie-recipes/mealie/blob/mealie-next/mealie/routes/households/controller_group_recipe_actions.py#L97).

As I see it, it would be necessary to pass in the fields (`yields`, `servings_amount`) from the client to the server, where it could attach the data to the post body.

@michael-genson I'd be willing to open a PR for that. Would that the aforementioned way to solve this work for you?

**Reply by felixschndr** [reply]
@hay-kot Sorry for the ping, I was unsure on how to get someones attention. Do you have any thoughts regarding this or shall I contact someone else?

**Reply by michael-genson** [reply]
Feel free to open a PR, then we can take a look!

**Reply by felixschndr** [reply]
PR opened https://github.com/mealie-recipes/mealie/pull/5340

**Comment by felixschndr** [comment]
FYI https://github.com/mealie-recipes/mealie/discussions/5006
